stubs/babel/babel/dates.pyi

```diff
@@ -1,30 +1,98 @@
-from datetime import date, datetime, time
-from typing import Any
+from datetime import date, datetime, time, timedelta, tzinfo
+from typing import Any, overload
+from typing_extensions import Literal
 
-NO_INHERITANCE_MARKER: str
-LC_TIME: Any
-date_ = date
-datetime_ = datetime
-time_ = time
+from babel.core import Locale
+from pytz import BaseTzInfo
+
+# The module contents here are organized the same way they are in the API documentation at
+# http://babel.pocoo.org/en/latest/api/dates.html
+
+# Date and Time Formatting
+def format_datetime(
+    datetime: datetime | None = ..., format: str = ..., tzinfo: tzinfo | None = ..., locale: str | Locale = ...
```

Comment:
Since the `format` argument also accepts any user-defined pattern, I originally decided to just use `str` since I thought it'd be redundant to use a union of some string literals and `str`. But I agree that it is nice to have those options listed out.